### ISSUE TITLE
Don't allow function parameters that overload keywords

### ISSUE THREAD

**Issue by erewok** [pellagic-puffbomb/servant-py#3]
We have a capture in one of our test API endpoints where the variable is named `id`. This is created as a function parameter and passed into the client function.

Ideally, we wouldn't overload any Python keywords, such as `id`, because it's gross and poor form.

Possibly check for membership in a list of keywords and rename if so.

**Comment by sboosali** [comment]
yeah, I wanted to do this too when I was playing with servant, but I don't know if it's possible with the operations supported on ghc's current type level strings. type level string comparison seems to exist, but I forget the details of why I didn't think this worked on my earlier attempt. so I ended up just checking things on the value level. 

https://hackage.haskell.org/package/base-4.10.0.0/docs/GHC-TypeLits.html#v:sameSymbol

**Comment by erewok** [comment]
I think this should be possible if we reify the strings and compare. I'll have to dig into a bit more. I'm still trying to figure out how to restructure the resulting code. Probably after the new year, I'll have a chance to start on the rewrite.
